pkgs/applications/networking/p2p/tremotesf/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, cmake
+, pkg-config
+, mkDerivation
+, fetchFromGitHub
+, qtbase
+, qttools
+, kwidgetsaddons
+, kwindowsystem
+, fmt
+, libpsl
+, cxxopts
+, wrapQtAppsHook
+}:
+
+mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here.

Since this is your first PR, let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Reply:
Thank you for the context/docs. Will tweak things to use finalAttrs & other fixes